### ISSUE TITLE
[css-page-3] Replace CSS 2.1 syntax definitions by CSS syntax

### ISSUE THREAD

**Issue by SebastianZ** [w3c/csswg-drafts#632]
The [minutes from 2016-10-12](https://lists.w3.org/Archives/Public/www-style/2016Oct/0091.html) resolved on removing the CSS grammar section from CSS 2.2 and instead refer to the CSS Syntax module. Though the CSS Paged Media Module still [uses that grammar for the definition of the @page rule](https://drafts.csswg.org/css-page-3/#syntax-page-selector).

This syntax should be replaced by something like this*:

```
@page <page-selector-list> { <page-body> }

where
<page-selector-list> = [ <page-selector># ]?
<page-body> = <declaration>? [ ; <page-body> ]? | <page-margin-box> <page-body>
<page-selector> = <pseudo-page>+ | <ident> <pseudo-page>*
<page-margin-box> = <page-margin-box-type> { <declaration-list> }
<pseudo-page> = : [ left | right | first | blank ]
<page-margin-box-type> = @top-left-corner | @top-left | @top-center | @top-right | @top-right-corner | @bottom-left-corner | @bottom-left | @bottom-center | @bottom-right | @bottom-right-corner | @left-top | @left-middle | @left-bottom | @right-top | @right-middle | @right-bottom
```

Sebastian

\* For reference, I used that [syntax definition for @page on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/@page#Formal_syntax).

**Comment by SebastianZ** [comment]
The previous syntax definition got replaced by one following the Syntax spec. in https://github.com/w3c/csswg-drafts/commit/30e1273058b3315ed4dfb2f74e26bf27e4eed0c2.

This change let parts still be defined in prose. @tabatkins As you made the change, would you agree to define everything in the formal syntax as I did back then?

Sebastian

**Comment by tabatkins** [comment]
I'm not sure I understand your question. The only difference between the grammar you suggested in the OP and what I eventually committed is that I used the guidelines in the Syntax spec for defining rules (which requires prose to specify what's allowed in a given `<definition-list>`), while your grammar tries to enforce the prose more strictly in the grammar itself. I intentionally wrote the Syntax guidelines to not do that, precisely because writing a correct and understandable grammar for block contents in various situations is difficult.
